src/pages/settings/AddLoginPage.js

```diff
@@ -0,0 +1,155 @@
+    }
+
+    submitForm() {
+        setSecondaryLogin(this.state.login, this.state.password)
+            .then((response) => {
+                if (response.jsonCode === 200) {
+                    Navigation.navigate(ROUTES.SETTINGS_PROFILE);
+                }
+            });
+    }
+
+    // Determines whether form is valid
+    validateForm() {
+        const validationMethod = this.formType === 'phone' ? Str.isValidPhone : Str.isValidEmail;
+        return !this.state.password || !validationMethod(this.state.login);
+    }
+
+    render() {
+        return (
+            <ScreenWrapper>
+                <HeaderWithCloseButton
+                    title={this.formType === 'phone' ? 'Add Phone Number' : 'Add Email Address'}
+                    shouldShowBackButton
+                    onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS_PROFILE)}
+                    onCloseButtonPress={Navigation.dismissModal}
+                />
+                <View style={[styles.p5, styles.flex1, styles.overflowScroll]}>
+                    <View style={styles.flexGrow1}>
+                        <Text style={[styles.mb6, styles.textP]}>
+                            {this.formType === 'phone'
+                                ? 'Enter your preferred phone number and password to send a validation link.'
+                                : 'Enter your preferred email address and password to send a validation link.'}
+                        </Text>
+                        <View style={styles.mb6}>
+                            <Text style={[styles.mb1, styles.formLabel]}>
+                                {this.formType === 'phone' ? 'Phone Number' : 'Email Address'}
+                            </Text>
+                            <TextInput
+                                style={styles.textInput}
+                                value={this.state.login}
+                                onChangeText={login => this.setState({login})}
+                                autoFocus
+                                keyboardType={this.formType === 'phone' ? 'phone-pad' : undefined}
+                                returnKeyType="done"
+                            />
+                        </View>
+                        <View style={styles.mb6}>
+                            <Text style={[styles.mb1, styles.formLabel]}>Password</Text>
+                            <TextInput
+                                style={styles.textInput}
+                                value={this.state.password}
+                                onChangeText={password => this.setState({password})}
+                                secureTextEntry
+                                autoCompleteType="password"
+                                textContentType="password"
+                                onSubmitEditing={this.submitForm}
+                            />
+                        </View>
+                        {!_.isEmpty(this.props.user.error) && (
```

Comment:
One thing I noticed is that this error message doesn't get formatted:
<img width="455" alt="Screen Shot 2021-03-30 at 7 59 10 PM" src="https://user-images.githubusercontent.com/12075600/113084284-89516e00-9192-11eb-9a4a-b7f12de9e796.png">

Reply:
Would I just be looking for specific text used in the message and splitting to use the first sentence or did you have something else in mind?

Reply:
> Would I just be looking for specific text used in the message and splitting to use the first sentence

Yeah, this was pretty much exactly my thought!